Add redirect_uri as optional param when getting Access Token

sirosen opened the pull request:
When getting an access token from code, OAuth2 spec dictates that the
redirect_uri is required.

This is still being tested, so not necessarily safe to merge yet.

sirosen committed:
Add redirect_uri as optional param when getting AT

When getting an access token from code, OAuth2 spec dictates that the
redirect_uri is required. Otherwise, validation should fail (and after the
upcoming Globus service changes, it will fail).

mattias-lidman commented:
Travis CI build failed...?

<rolls dice>

mattias-lidman commented:
:shipit:

bjmc commented:
Shouldn't this take a `redirect_uri` for the first step getting an authorization code, too?

sirosen committed:
Add redirect_uri option to URL generation

The utility function which generates the authorize URI (parameterized over user
data) now accepts a redirect_uri as a kwarg, rather than expecting the user to
correctly append it. This better integrates with the new support for
redirect_uris on token exchanges.

sirosen commented:
I think we need to expect the Travis builds to fail -- they fail on the `integration` branch as well.
A lot of the infra surrounding this repo seems quite out of date and in need of an overhaul.

Per Brendan's point, I just added redirect_uri support to the segment that generates the authorize URI from inputs. Hopefully we'll be able to validate that this works shortly.

bjmc commented:
Sure :horse: :sheep: :horse_racing:

sirosen committed:
Update version to 0.0.2, update Readme

Minor tweak to Readme to make it clearer that this is just an example client.
Fix readme to say "Globus" not "GlobusOnline".
Increment version number.

sirosen commented:
I've gone ahead and merged this, since it passed testing done internally.

As per an out of band discussion, I've also taken the opportunity to bump the version to 0.0.2
